crates/turborepo-repository/src/discovery.rs

```diff
@@ -0,0 +1,123 @@
+//! turborepo-discovery
+//!
+//! This package contains a number of strategies for discovering various things
+//! about a workspace. These traits come with a basic implementation and some
+//! adaptors that can be used to compose them together.
+//!
+//! This powers various intents such as 'query the daemon for this data, or
+//! fallback to local discovery if the daemon is not available'. Eventually,
+//! these strategies will implement some sort of monad-style composition so that
+//! we can track areas of run that are performing sub-optimally.
+
+use tokio_stream::{iter, StreamExt};
+use turbopath::AbsoluteSystemPathBuf;
+
+use crate::package_manager::PackageManager;
+
+#[derive(Clone)]
+pub struct WorkspaceData {
+    pub package_json: AbsoluteSystemPathBuf,
+    pub package_manager: PackageManager,
```

Comment:
This feels slightly off that `package_manager` is part of `WorkspaceData`. Is there ever a reason where a call to `discover_packages` would return a vec with differing `package_manager`s?